lib/hirefire/macro/sidekiq.rb

```diff
@@ -58,6 +58,35 @@ def queue(*queues)
 
         [in_queues, in_schedule, in_retry, in_progress].compact.inject(&:+)
       end
+
+      # Reports the latency (in seconds) for the provided Sidekiq queue(s).
+      #
+      # @example Sidekiq Macro Usage
+      #   HireFire::Macro::Sidekiq.latency # average latency of all queues
+      #   HireFire::Macro::Sidekiq.latency("email") # only email queue
+      #   HireFire::Macro::Sidekiq.latency("audio", "video") # average of audio and video queue latencies
+      #   HireFire::Macro::Sidekiq.latency("email", "video", weighted: true) # weighted average of latencies
+      def latency(*queues)
+        require "sidekiq/api"
+
+        queues.flatten!
+
+        if queues.last.is_a?(Hash)
+          options = queues.pop
+        else
+          options = {}
+        end
+
+        queues = queues.map(&:to_s)
+        queues = ::Sidekiq::Stats.new.queues.map { |name, _| name } if queues.empty?
+
+        latencies = if options[:weighted]
+          queues.map {|queue| q = ::Sidekiq::Queue.new(queue); q.latency * q.size }
```

Comment:
The weighted average algorithm is not quite correct, I think. It should use the sum of the queue sizes as the divisor, not the number of queues measured (`queues.length`)